### PR TITLE
Use TaskSpec instead of TaskTemplate for fetch_task and avoid network when loading module

### PR DESCRIPTION
**Pull request by kumare3**
Signed-off-by: Ketan Umare <ketan.umare@gmail.com>

# TL;DR
This PR converts a remote taskClosure and WorkflowClosure to a Spec (with default top level attributes). 
This helps in
- Simplified handling. 
- Local tasks and workflows are unified with remote
- removes any complex handling

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3176

### CODE REVIEW

File: flytekit/tools/translator.py [hunk 1]
```diff
@@ -563,6 +562,52 @@ def get_reference_spec(
     return ReferenceSpec(template)
 
 
+def get_serializable_flyte_workflow(entity: FlyteWorkflow, settings: SerializationSettings) -> FlyteControlPlaneEntity:
```
Comment (kumare3) [inline flytekit/tools/translator.py:565]: @wild-endeavor take a look at this part

**Review by wild-endeavor — DISMISSED** [review]

**Review by wild-endeavor — APPROVED** [review]
